evm_loader/program/src/precompile_contracts.rs

```diff
@@ -301,22 +308,36 @@ const QUERY_ACCOUNT_METHOD_DATA_ID: &[u8; 4] = &[0x43, 0xca, 0x51, 0x61];
 
 /// Call inner `query_account`
 #[must_use]
+#[allow(clippy::too_many_lines)]
 pub fn query_account<'a, B: AccountStorage>(
     input: &[u8],
     state: &mut ExecutorState<'a, B>
 )
     -> Capture<(ExitReason, Vec<u8>), Infallible>
 {
     let (method_id, rest) = input.split_at(4);
-    let method_id: &[u8; 4] = method_id.try_into().unwrap_or_else(|_| &[0_u8; 4]);
+    let method_id: &[u8; 4] = method_id.try_into().unwrap_or(&[0_u8; 4]);
     let (account_address, rest) = rest.split_at(32);
     let account_address = Pubkey::new(account_address);
 
     match method_id {
+        QUERY_ACCOUNT_METHOD_CACHE_ID => {
+            // Note: abi.encodeWithSignature makes parameters padded to 32 bytes
+            let (offset, rest) = rest.split_at(32);
+            let (length, _) = rest.split_at(32);
+            let offset = U256::from_big_endian_fast(offset).as_usize();
+            let length = U256::from_big_endian_fast(length).as_usize();
```

Comment:
Check how arguments are parsed in `fn erc20_wrapper`.
Having the same things implemented differently in different places makes code harder to read.

Reply:
Done.